src/frontend/src/scheduler/hummock_snapshot_manager.rs

```diff
@@ -278,11 +281,25 @@ impl HummockSnapshotManagerCore {
         // Check the min epoch which still exists running query. If there is no running query,
         // we shall unpin snapshot with the last committed epoch.
         let min_epoch = match self.epoch_to_query_ids.first_key_value() {
-            Some((epoch, _)) => *epoch,
+            Some((epoch, query_ids)) => {
+                assert!(
+                    !query_ids.is_empty(),
+                    "No query is associated with epoch {}",
+                    epoch
+                );
+                *epoch
+            }
             None => last_committed_epoch,
         };
 
-        if min_epoch <= self.last_unpin_snapshot.load(Ordering::Acquire) {
+        let last_unpin_snapshot = self.last_unpin_snapshot.load(Ordering::Acquire);
+        tracing::info!(
```

Comment:
This log info is too frequently....

Reply:
> This log info is too frequently....

One log entry every UNPIN_INTERVAL_SECS (10sec) seems fine for me. Btw, we have a log entry for unpin rpc in L310.

Reply:
How about only log it when `min_epoch > last_unpin_snapshot`?

Reply:
> This log info is too frequently....

Changed to log it in trace level.